File: trunk/base/doc/guides/CreateTableExample.py
#!/usr/bin/env python

import sys
import rgma

vdb = "default"
create = "create table userTable (userId VARCHAR(255) NOT NULL " \
         "PRIMARY KEY, aString VARCHAR(255), aReal REAL, anInt INTEGER)"
rule = "::RW"

try:
    rgma.Schema(vdb).createTable(create, rule)
    
except rgma.RGMAException, e:
    sys.stderr.write("RGMAException " +  e.getMessage() + "\n")
    sys.exit(1)

File: trunk/base/doc/guides/CreateTableExample.c
#include <stdio.h>
#include <stdlib.h>

#include "rgma/rgma.h"

int main(int argc, char *argv[]) {
    char *vdb = "default";
    char *create = "create table userTable (userId VARCHAR(255) NOT NULL \
           PRIMARY KEY, aString VARCHAR(255), aReal REAL, anInt INTEGER)";
    char *rules[] = { "::RW" };
    int numRules = 1;
    RGMAException *exception;

    RGMASchema_createTable(vdb, create, numRules, rules, &exception);
    if (exception) {
        fprintf(stderr, "Failed to create table %s\n", exception->message);
        RGMA_freeException(exception);
        exit(1);
    }
    exit(0);
}
